### PR TITLE
Changes to integration tests to use OCR

### PR DESCRIPTION
**Pull request by sankarpn**
Fix the integration tests to use OCR instead of docker store for WLS images.

### CODE REVIEW

File: integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java [hunk 1]
```diff
@@ -69,6 +69,8 @@
   protected static String appLocationInPod = "/u01/oracle/apps";
   private static String appLocationOnHost;
   private static Properties appProps;
+  private static String imageTag;
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java]: change imageTag to weblogicImageTag

File: integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java [hunk 2]
```diff
@@ -69,6 +69,8 @@
...
+  private static String imageName;
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java]: change imageName to weblogicImageName

File: integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java [hunk 3]
```diff
@@ -104,6 +106,8 @@ public static void initialize(String appPropsFile) throws Exception {
     }
     username = appProps.getProperty("username", username);
     password = appProps.getProperty("password", password);
+    imageTag = appProps.getProperty("weblogicImageTag");
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java]: Environment variables can override the default value set in props, replace with this
```
weblogicImageTag = System.getenv("IMAGE_TAG_WEBLOGIC") != null ? System.getenv("IMAGE_TAG_WEBLOGIC")  : appProps.getProperty("weblogicImageTag");
weblogicImageName = System.getenv("IMAGE_NAME_WEBLOGIC") != null ? System.getenv("IMAGE_NAME_WEBLOGIC")  : appProps.getProperty("weblogicImageName");
```

File: integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java [hunk 4]
```diff
@@ -227,6 +231,23 @@ public static void initialize(String appPropsFile) throws Exception {
     logger.info("Env var BRANCH_NAME " + System.getenv("BRANCH_NAME"));
   }
 
+  /**
+   * getter method for imageTag field
+   *
+   * @return image tag of the WLS docker images
+   */
+  public static String getImageTag() {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java]: change to getWeblogicImageTag()

File: integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java [hunk 5]
```diff
@@ -227,6 +231,23 @@ public static void initialize(String appPropsFile) throws Exception {
...
+    return imageTag;
+  }
+
+  /**
+   * getter method for imageName
+   *
+   * @return image name of the WLS docker image
+   */
+  public static String getImageName() {
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java]: change to getWeblogicImageName()

File: integration-tests/src/test/resources/setupenv.sh [hunk 6]
```diff
@@ -87,7 +87,7 @@ function pull_tag_images {
 	  echo "Creating Docker Secret"
 	  
 	  kubectl create secret docker-registry $IMAGE_PULL_SECRET_WEBLOGIC  \
-	    --docker-server=index.docker.io/v1/ \
+	    --docker-server=container-registry.oracle.com/ \
```
Comment (vanajamukkara) [inline integration-tests/src/test/resources/setupenv.sh]: Add a variable WL_DOCKER_SERVER and use that in the script

File: integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java (hunk 1 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java]: Done

File: integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java (hunk 2 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java]: Done

File: integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java (hunk 3 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java]: Done

File: integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java (hunk 4 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java]: Done

File: integration-tests/src/test/resources/setupenv.sh (hunk 6 above)
Comment (sankarpn) [reply integration-tests/src/test/resources/setupenv.sh]: Done

**Comment by vanajamukkara** [comment]
Looks good. Please post both internal and external jenkins results once you have...

**Comment by sankarpn** [comment]
Local runs. The ITMonitoringExporter tests are failing in develop branch as well.


[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   ITMonitoringExporter.test14_ChangeConfigNoCredentials:452
[ERROR] Errors:
[ERROR]   ITMonitoringExporter.test02_ReplaceConfiguration:152->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test03_AppendConfiguration:201->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test04_ReplaceOneAttributeValueAsArrayConfiguration:232->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test05_AppendArrayWithOneExistedAndOneDifferentAttributeValueAsArrayConfiguration:253->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test06_ReplaceWithEmptyConfiguration:272->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test07_AppendWithEmptyConfiguration:292->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test08_1AppendWithNotYmlConfiguration:309->changeConfigNegative:545->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test08_2ReplaceWithNotYmlConfiguration:326->changeConfigNegative:545->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test10_ReplaceWithCorruptedYmlConfiguration:361->changeConfigNegative:545->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test11_ReplaceWithDublicatedValuesConfiguration:380->changeConfigNegative:545->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test12_AppendWithDublicatedValuesConfiguration:399->changeConfigNegative:545->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test13_ReplaceMetricsNameSnakeCaseFalseConfiguration:423->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[INFO]
[ERROR] Tests run: 69, Failures: 1, Errors: 12, Skipped: 0
[INFO]
[INFO]
[INFO] --- maven-failsafe-plugin:3.0.0-M3:verify (integration-tests) @ operator-integration-tests ---
[INFO] Failsafe report directory: /scratch/speriyat/weblogic-kubernetes-operator/integration-tests/target/failsafe-reports
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Build Tools 1.0 .................................... SUCCESS [  2.254 s]
[INFO] weblogic-kubernetes-operator 2.2 ................... SUCCESS [  5.393 s]
[INFO] json-schema 2.2 .................................... SUCCESS [ 14.191 s]
[INFO] jsonschema-maven-plugin Maven Mojo 2.2 ............. SUCCESS [ 12.938 s]
[INFO] operator-model 2.2 ................................. SUCCESS [ 25.114 s]
[INFO] operator-swagger 2.2 ............................... SUCCESS [  6.126 s]
[INFO] operator-runtime 2.2 ............................... SUCCESS [ 58.254 s]
[INFO] operator-integration-tests 2.2 ..................... FAILURE [  03:27 h]
[INFO] installation-tests 2.2 ............................. SKIPPED
[INFO] Project Reports 2.2 ................................ SKIPPED
[INFO] ------------------------------------------------------------------------

**Comment by sankarpn** [comment]
The Jenkins and local tests passed except the ITMonitoringExporter. It looks like it is failing in develop also.

http://wls-jenkins.us.oracle.com/view/weblogic-operator/job/weblogic-kubernetes-operator-javatest/1705/console


05-29-2019 16:48:18 INFO   oracle.kubernetes.operator.ITUsabilityOperatorHelmChart staticUnPrepare SUCCESS
[INFO] Tests run: 13, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1,687.178 s - in oracle.kubernetes.operator.ITUsabilityOperatorHelmChart
[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   ITMonitoringExporter.test14_ChangeConfigNoCredentials:452
[ERROR] Errors:
[ERROR]   ITElasticLogging.staticPrepare:62 » Runtime FAILURE: command cd /scratch/speri...
[ERROR]   ITMonitoringExporter.test02_ReplaceConfiguration:152->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test03_AppendConfiguration:201->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test04_ReplaceOneAttributeValueAsArrayConfiguration:232->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test05_AppendArrayWithOneExistedAndOneDifferentAttributeValueAsArrayConfiguration:253->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test06_ReplaceWithEmptyConfiguration:272->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test07_AppendWithEmptyConfiguration:292->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test08_1AppendWithNotYmlConfiguration:309->changeConfigNegative:545->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test08_2ReplaceWithNotYmlConfiguration:326->changeConfigNegative:545->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test10_ReplaceWithCorruptedYmlConfiguration:361->changeConfigNegative:545->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test11_ReplaceWithDublicatedValuesConfiguration:380->changeConfigNegative:545->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test12_AppendWithDublicatedValuesConfiguration:399->changeConfigNegative:545->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[ERROR]   ITMonitoringExporter.test13_ReplaceMetricsNameSnakeCaseFalseConfiguration:423->submitConfigureForm:573->submitConfigureForm:608 » FailingHttpStatusCode
[INFO]
[ERROR] Tests run: 67, Failures: 1, Errors: 13, Skipped: 0
[INFO]
[INFO]
[INFO] --- maven-failsafe-plugin:3.0.0-M3:verify (integration-tests) @ operator-integration-tests ---
[INFO] Failsafe report directory: /scratch/speriyat/weblogic-kubernetes-operator/integration-tests/target/failsafe-reports
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Build Tools 1.0 .................................... SUCCESS [  2.187 s]
[INFO] weblogic-kubernetes-operator 2.2 ................... SUCCESS [  5.367 s]
[INFO] json-schema 2.2 .................................... SUCCESS [ 14.020 s]
[INFO] jsonschema-maven-plugin Maven Mojo 2.2 ............. SUCCESS [  9.983 s]
[INFO] operator-model 2.2 ................................. SUCCESS [ 21.982 s]
[INFO] operator-swagger 2.2 ............................... SUCCESS [  5.404 s]
[INFO] operator-runtime 2.2 ............................... SUCCESS [ 52.607 s]
[INFO] operator-integration-tests 2.2 ..................... FAILURE [  03:22 h]
[INFO] installation-tests 2.2 ............................. SKIPPED
[INFO] Project Reports 2.2 ................................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:24 h

**Pull request merged**